src/main/drivers/exti.c

```diff
@@ -184,6 +184,38 @@ void EXTIEnable(IO_t io, bool enable)
 #endif
 }
 
+void EXTISetTrigger(IO_t io, EXTITrigger_TypeDef trigger)
+{
+uint32_t extiMask;
+
+#if defined(STM32F1) || defined(STM32F4) || defined(STM32F7)
```

Comment:
RTSR , FTSR and IMR are identical (F3 has more that 32 EXTI sources and more registers, but IO EXTI is in same place). The target-specific part can be moved into EXTIMask(IO_t) function and shared.